src/support/err.c

```diff
@@ -504,12 +502,11 @@ __wt_illegal_value(WT_SESSION_IMPL *session, const char *name)
 	    name == NULL ? "" : name, name == NULL ? "" : ": ",
 	    "encountered an illegal file format or internal value");
 
-#if !defined(HAVE_DIAGNOSTIC)
-	return (__wt_panic(session));
+#if defined(HAVE_DIAGNOSTIC)
+	__wt_abort(session);                    /* Drop core if testing. */
 #endif
 
-	__wt_abort(session);			/* Drop core if testing. */
-	/* NOTREACHED */
+	return (WT_ERROR);
```

Comment:
@keithbostic, `__wt_illegal_value` still returns an error, the only code that will run afterwards is error handling.  In this case, a block read should fail, but that shouldn't bring down the system: it just means there is a file we can't open.  That seems very different to me to a programming error where we control all of the possible values that could go into a `switch`.

My concern about the panic here (and more generally, the number of calls to panic through the code) is that I can't reason about whether an operation might cause a panic or not.  In this case, that means it's impossible to handle this failure in a more graceful way.  It isn't possible for MongoDB to figure out what has caused this kind of failure short of parsing WiredTiger's error log...